sdk/resourcemanager/Azure.ResourceManager/src/ArmClientOptions.cs

```diff
@@ -35,5 +38,48 @@ public void SetApiVersion(ResourceType resourceType, string apiVersion)
 
             ResourceApiVersionOverrides[resourceType] = apiVersion;
         }
+
+        /// <summary>
+        /// Sets the api versions from an Azure Stack profile.
+        /// </summary>
+        public void SetApiVersionsFromProfile(AzureStackProfile profile)
+        {
+            var assembly = Assembly.GetExecutingAssembly();
+            var profileFileFullName = $"Azure.ResourceManager.Assets.Profile.{profile.GetFileName()}";
```

Comment:
should we just change this to get `GetManifestName` and put this additional logic inside the extension method as well?